crates/fuel-core/src/service/adapters.rs

```diff
@@ -83,19 +83,54 @@ impl P2PAdapter {
     }
 }
 
+#[cfg(feature = "p2p")]
+#[async_trait::async_trait]
+impl NetworkInfo for P2PAdapter {
+    async fn connected_reserved_peers(&self) -> usize {
+        if let Some(service) = self.service.as_ref() {
+            service.connected_reserved_peers_count().await.unwrap_or(0)
+        } else {
+            0
+        }
+    }
+}
+
 #[cfg(not(feature = "p2p"))]
 impl P2PAdapter {
     pub fn new() -> Self {
         Default::default()
     }
 }
 
-pub struct SyncAdapter {
+#[cfg(not(feature = "p2p"))]
+#[async_trait::async_trait]
+impl NetworkInfo for P2PAdapter {
+    async fn connected_reserved_peers(&self) -> usize {
+        // returns fake number of connected peers - since p2p is not enabled at all
+        usize::MAX
```

Comment:
I would prefer to add a new additional function, `is_enable`, that indicates whether p2p is enabled or not. In this case, we can return here 0, but on the PoA side, we can handle it.

Also, if p2p is disabled then sync service is disabled and we can use old logic=)